### ISSUE TITLE
Onlinemode Automated Offlinemode

### ISSUE THREAD

**Issue by Sabermc7** [TotalFreedom/TotalFreedomMod#1026]
This has been a big issue going on for quite a while. Each day, the server is left on onlinemode forever. By default, TotalFreedom is a cracked server. But when someone enables onlinemode, they're forgetting to turn it off afterwards. Today the server has been on onlinemode for several hours and there isn't a specific reason given.

So this suggestion is, every 30 minutes if onlinemode is left on or so, the server will automatically turn off onlinemode. Because it's like adminmode left on forever if someone turns it on.

Before anyone who objects says "just call out in the shoutbox", no. Most of us cracked user admins have been constantly requesting in the shoutbox to turn onlinemode off but not resolved, several hours later... even if shoutbox isn't cleared. Not even tweeting to @TotalFreedomMC helps, still no sign of anyone making server back to cracked.

**Comment by JeromSar** [comment]
Good idea @Sabermc7

**Comment by Wild1145** [comment]
I disagree, there have been plenty of times where onlinemode has been required for an extended period of time and this would make it impossible to keep onlinemode true for more than 30mins, thus causing the potential for spambots that we block with onlinemode to then join therefore crashing the server.

**Comment by ReelPattytothewatty** [comment]
Have you seen all of the admins we have! One of those admins could look at the shoutbox and onlinemode off. Simple.

**Comment by JeromSar** [comment]
How about an option to set this timer automatically?

**Comment by OnlineRepz** [comment]
@ReelPattytothewatty, well, online mode is blocked to console (Telnet)...

**Comment by ReelPattytothewatty** [comment]
@OnlineRepz True, True.

**Comment by SnowGamerMC** [comment]
Ideekei if jejejfjejdjdjddjdjeejejdddjrjfd

**Comment by Sabermc7** [comment]
@ReelPattytothewatty did you not read everything I said? Even if you request in the shoutbox, still no sign of onlinemode off. A few days ago most of us supers have been requesting like every hour yet no one seems to get the message, it was left on for several hours.

@Wild1145, maybe a telnet could just enable it every 30 minutes when it gets turned off if there's a big issue. So that they wont forget.

**Comment by ReelPattytothewatty** [comment]
@Sabermc7 Well, maybe they should get a REAL account. It's unfair for people who are premium to play with cracked users. Premium should get some more time with premiums. That way, they'll barely be no griefers.

**Comment by Sabermc7** [comment]
Umm no, @ReelPattytothewatty that sounds very ignorant, it sounds like to me that you hate cracked users and you want to banish them off the server forever, it doesn't matter if someone is premium or not, this server by default is suppose to be cracked, we do not enable onlinemode just to let premiums have time together, that's disrespectful to cracked users, also who said anything about premium members being **forced** to play with cracked users? They can just ignore them and move on, not banish them off the server. This is also a "Freedom" server, and by keeping the server premium for like 12 hours just gets rid of the freedom.

**Comment by JeromSar** [comment]
No, whether this issue will be implemented or not does not depend on the difference between cracked and non-cracked players @ReelPattytothewatty.

**Comment by Sabermc7** [comment]
Anyways, i'll close this issue now as it's accepted and to prevent going off-topic. :)

**Comment by Sabermc7** [comment]
:)

**Comment by ChargedCreeper** [comment]
Great idea. I can't tell you how many times I have had to disable onlinemode because of this issue.

**Comment by supergameplayer** [comment]
We can also do a custom timer for onlinemode. (/onlinemode on 15 would turn on online mode for 15 minutes)

**Comment by RobinGall2910** [comment]
I'd go with @supergameplayer's idea too :P

**Comment by FilezPlays** [comment]
@supergameplayer's idea sounds pretty good. Go for gold!
